### PR TITLE
Feature/csv2ctrlsig

### PR DESCRIPTION
**Pull request by kasettli**
Add a utility function that generates a control signal from a CSV file.

<a href="https://gitpod.io/#https://github.com/hammal/cbadc/pull/67"></a>

### CODE REVIEW

**Comment by kasettli** [comment]
@hammal please also have a look at the exception that is raised. I'm not that familiar with this.

File: src/cbadc/utilities.py [hunk 1]
```diff
@@ -232,6 +232,54 @@ def read_byte_stream_from_url(
...
+    """Creates an iterator that reads from a CSV file.  
+
+    Parameters
+    ----------
+    filename : `str`
+        filename for input file
+    M : `int`
+        number of controls
+    msb2lsb : `bool`
+        set order of control bits. Default is LSB to MSB
```
Comment (hammal) [inline src/cbadc/utilities.py:245]: let's write out the abbreviations.

least significant bit (LSB)

File: src/cbadc/utilities.py [hunk 2]
```diff
@@ -232,6 +232,54 @@ def read_byte_stream_from_url(
...
+
+    Yields
+    ------
+    array_like
+        a control signal sample
+
+    Example
+    -------
+    >>> # Given the following file, named `test.csv`, for a setup with four control bits (M=4):
```
Comment (hammal) [inline src/cbadc/utilities.py:254]: can we create this file in the same doctext example something like

```
with open('test.csv','w') as f:
    f.write("""
0,0,0,0
0,0,0,1
0,0,1,1
0,1,1,1
""")
```

takes as long as the comment above...

File: src/cbadc/utilities.py [hunk 3]
```diff
@@ -232,6 +232,54 @@ def read_byte_stream_from_url(
...
+    ------
+    `RuntimeError`
+        for lines with number of entries different from M.
+        
+    """
+    with open(filename, 'r') as read_obj:
+        for line in read_obj:
+            s = np.fromstring(line, dtype=np.bool, sep=',')
+            if (s.size != M):
+                raise RuntimeError("The number of entries in the current line is not equal to M")
```
Comment (hammal) [inline src/cbadc/utilities.py:279]: This is a good way to use RunTimeError!

**Review by hammal — CHANGES_REQUESTED** [review]
So except for the proposed changes looks good to me!

File: src/cbadc/utilities.py (hunk 1 above)
Comment (kasettli) [reply src/cbadc/utilities.py]: good point

File: src/cbadc/utilities.py (hunk 2 above)
Comment (kasettli) [reply src/cbadc/utilities.py]: ah good idea. I was wondering how to do this in a neat way.

File: src/cbadc/utilities.py (hunk 3 above)
Comment (kasettli) [reply src/cbadc/utilities.py:283]: ok, thx

File: src/cbadc/utilities.py [hunk 4]
```diff
@@ -232,6 +232,58 @@ def read_byte_stream_from_url(
...
+
+    Parameters
+    ----------
+    filename : `str`
+        filename for input file
+    M : `int`
+        number of controls
+    msb2lsb : `bool`
+        bit order of input file.
+        Default is least significant to most significant bit from left to right (LSB to MSB).
```
Comment (hammal) [inline src/cbadc/utilities.py:246]: Default is least significant bit (LSB) to most significant bit (MSB) from left to right.

**Review by hammal — CHANGES_REQUESTED** [review]

File: src/cbadc/utilities.py (hunk 4 above)
Comment (kasettli) [reply src/cbadc/utilities.py]: yes, that sounds a bit clearer

**Review by hammal — APPROVED** [review]
Shiny!

**Pull request merged**